static/app/components/replays/player/replayPlayerContainment.tsx

```diff
@@ -0,0 +1,84 @@
+import type {CSSProperties, ReactNode} from 'react';
+import {useRef} from 'react';
+import {css} from '@emotion/react';
+
+import divide from 'sentry/utils/number/divide';
+import useReplayPlayerState from 'sentry/utils/replays/playback/providers/useReplayPlayerState';
+import {useDimensions} from 'sentry/utils/useDimensions';
+
+interface Props {
+  /**
+   * You must pass `styles` into the <ReplayPlayer>,
+   */
+  children: (styles: CSSProperties) => ReactNode;
+
+  /**
+   * How to measure and resize the player.
+   *
+   * "both"
+   *    The default. Measure the player fully. Resize the player to fit within
+   *    the given width & height. Useful if you want to reserve some space for
+   *    the replay.
+   *    ie: with a wrapper that has CSS `height` set.
+   *
+   * "width"
+   *    Height available will not be measured. The replay will be
+   *   resized to accomodate the width available, and the height grows to
+   *   maintain the aspect-ratio. This can result in really tall replays when
+   *   captured on a mobile device.
+   *
+   * There is no "height" option, document flow doesn't usually expand in an
+   * uncontrolled way like height does.
+   */
+  measure?: 'both' | 'width';
+}
+
+export default function ReplayPlayerContainment({children, measure = 'both'}: Props) {
```

Comment:
Maybe a name related to what it's doing irt dimensions

```suggestion
export default function ReplayPlayerSizer({children, measure = 'both'}: Props) {
```

```suggestion
export default function ReplayPlayerMeasurer({children, measure = 'both'}: Props) {
```

```suggestion
export default function ReplayPlayerDimension({children, measure = 'both'}: Props) {
```